### PR TITLE
Support relations requirement in process syntax

### PR DESCRIPTION
**Pull request by mstajdohar**

### CODE REVIEW

File: resolwe/flow/static/flow/processSchema.json [hunk 1]
```diff
@@ -160,12 +171,45 @@
...
+                                "type": "array",
+                                "items": {
+                                    "type": "string"
+                                }
+                            },
+                            "required": {
+                                "type": "boolean"
+                            }
+                        },
+                        "required": [
```
Comment (hadalin) [inline resolwe/flow/static/flow/processSchema.json]: Should `category` also be required?

File: resolwe/flow/static/flow/processSchema.json [hunk 2]
```diff
@@ -56,12 +56,19 @@
         "persistence": {
             "description": "Processor data persistence",
             "type": "string",
-            "enum": ["RAW", "CACHED", "TEMP"]
+            "enum": [
```
Comment (dblenkus) [inline resolwe/flow/static/flow/processSchema.json]: Move all changes except `"relations"` to a separate commit.

File: resolwe/flow/static/flow/processSchema.json (hunk 1 above)
Comment (mstajdohar) [reply resolwe/flow/static/flow/processSchema.json]: No, see this use case: https://github.com/genialis/resolwe-bio/pull/740/files#diff-8abd821c113e3327b5dc4e1038345c3eR14

File: resolwe/flow/static/flow/processSchema.json (hunk 2 above)
Comment (mstajdohar) [reply resolwe/flow/static/flow/processSchema.json]: Those are from the autoformatter. I'll remove them.

Comment (mstajdohar) [reply resolwe/flow/static/flow/processSchema.json]: Done

File: resolwe/flow/static/flow/processSchema.json [hunk 3]
```diff
@@ -160,6 +160,33 @@
...
                     }
+                },
+                "relations": {
+                    "description": "List of relations used by the process",
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "properties": {
+                            "type": {
+                                "type": "string"
```
Comment (hadalin) [inline resolwe/flow/static/flow/processSchema.json]: @mstajdohar How about changing this type to `"enum": ["compare", "group", "background"]` to make it even more narrow?

**Pull request merged**